torchvision/prototype/datasets/_builtin/cub.py

```diff
@@ -0,0 +1,233 @@
+
+        return dict(
+            (self._2011_decode_ann if year == "2011" else self._2010_decode_ann)(anns_data, decoder=decoder),
+            image=decoder(buffer) if decoder else buffer,
+            label=Label(int(label_str), category=category),
+        )
+
+    def _make_datapipe(
+        self,
+        resource_dps: List[IterDataPipe],
+        *,
+        config: DatasetConfig,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+    ) -> IterDataPipe[Dict[str, Any]]:
+        if config.year == "2011":
+            archive_dp, segmentations_dp = resource_dps
+            images_dp, split_dp, image_files_dp, bounding_boxes_dp = Demultiplexer(
+                archive_dp, 4, self._2011_classify_archive, drop_none=True, buffer_size=INFINITE_BUFFER_SIZE
+            )
+
+            image_files_dp = CSVParser(image_files_dp, dialect="cub200")
+            image_files_map = dict(
+                (image_id, rel_posix_path.rsplit("/", maxsplit=1)[1]) for image_id, rel_posix_path in image_files_dp
+            )
+
+            split_dp = CSVParser(split_dp, dialect="cub200")
+            split_dp = Filter(split_dp, functools.partial(self._2011_filter_split, split=config.split))
+            split_dp = Mapper(split_dp, getitem(0))
+            split_dp = Mapper(split_dp, image_files_map.get)
+
+            bounding_boxes_dp = CSVParser(bounding_boxes_dp, dialect="cub200")
+            bounding_boxes_dp = Mapper(bounding_boxes_dp, image_files_map.get, input_col=0)
+
+            anns_dp = IterKeyZipper(
+                bounding_boxes_dp,
+                segmentations_dp,
+                key_fn=getitem(0),
+                ref_key_fn=self._2011_segmentation_key,
+                keep_key=True,
+                buffer_size=INFINITE_BUFFER_SIZE,
+            )
+        else:  # config.year == "2010"
+            split_dp, images_dp, anns_dp = resource_dps
+
+            split_dp = Filter(split_dp, path_comparator("stem", config.split))
+            split_dp = LineReader(split_dp, decode=True, return_path=False)
+            split_dp = Mapper(split_dp, self._2010_split_key)
+
+            anns_dp = Mapper(anns_dp, self._2010_anns_key)
+
+        split_dp = hint_sharding(split_dp)
+        split_dp = hint_shuffling(split_dp)
+
+        dp = IterKeyZipper(
+            split_dp,
+            images_dp,
+            getitem(),
+            path_accessor("name"),
+            buffer_size=INFINITE_BUFFER_SIZE,
+        )
+        dp = IterKeyZipper(
+            dp,
+            anns_dp,
+            getitem(0),
+            buffer_size=INFINITE_BUFFER_SIZE,
+        )
+        return Mapper(dp, functools.partial(self._collate_and_decode_sample, year=config.year, decoder=decoder))
+
+    def _generate_categories(self, root: pathlib.Path) -> List[str]:
+        config = self.info.make_config(year="2011")
+        resources = self.resources(config)
+
+        dp = resources[0].load(pathlib.Path(root) / self.name)
```

Comment:
Similarly to the changes in https://github.com/pytorch/vision/pull/5188, should this just be

```suggestion
        dp = resources[0].load(root)
```
?

Reply:
Yes, good catch.